pkg/aws/policies.go

```diff
@@ -241,7 +251,17 @@ func (c *awsClient) EnsurePolicy(policyArn string, document string,
 		if aerr, ok := err.(awserr.Error); ok {
 			switch aerr.Code() {
 			case iam.ErrCodeNoSuchEntityException:
-				return c.createPolicy(policyArn, document, tagList, path)
+				policyArn, err = c.createPolicy(policyArn, document, tagList, path)
+				if err != nil {
+					if aerr, ok := err.(awserr.Error); ok {
+						if aerr.Code() == iam.ErrCodeEntityAlreadyExistsException {
```

Comment:
I am just wondering, how do we get to this exception? 
If we only trigger the create policy when no such entity exists, but yet we try handle entity already exists exception. Maybe I am missing something, so just looking for some clarification

Reply:
AWS allows you to retrieve a policy by providing a full ARN. We have the wrong path so we have a non-existing ARN. However, the policy name is the unique key. Creating a policy uses a different input method. It uses name and path. So 'create policy' will fail with 'already exists'. This is the indication that the ARN we have is wrong. Probably wrong path.

Reply:
Damn that is annoying, that we can not search with the came criteria that we are creating with. Unless we fetched all policies and filtered them. Guess there is some overhead with that.

Reply:
indeed